sycl/doc/extensions/proposed/sycl_ext_oneapi_fpga_mem.asciidoc

```diff
@@ -0,0 +1,507 @@
+
+* `mlab`: data is stored in special Adaptive Logic Modules (ALMs) called memory-logic array blocks 
+* `block_ram`: data is stored in memory blocks, ie. M20Ks
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+num_banks
+----
+a|
+Number of banks that the array is divided into. If `stride_size` property is not specified in conjunction with `num_banks` property, memory will be divided using cyclic partitioning strategy, where consecutive words will be placed in different banks.
+
+Must be greater than zero and less than number of array elements.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+stride_size
+----
+a|
+Number of consecutive elements in an array that will be part of the same bank. If `num_banks` property is not specified in conjunction with `stride_size` property, memory will be divided using block partitioning strategy, where consecutive words will be placed in the same bank until `stride_size` worth of elements is reached, at which point further words will be added to a new bank.
+
+Must be greater than zero and less than number of array elements.
+
+If `word_size` property is specified, `stride_size` must be a multiple of `word_size`. 
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+word_size
+----
+a|
+Size in elements of a single memory transaction.
+
+Must be a factor of number of array elements.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+port_direction
+----
+a|
+The following values are supported:
+
+* `uni`: Specifies that a port to the memory can only service read or write requests, but not both.
+* `bi`: Specifies that a port to the memory can service both read and write requests.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+clock_2x
+----
+a|
+The following values are supported:
+
+* `false`: Specifies that the memory implementing the variable must operate at the same clock frequency as the kernel accessing it.
+* `true`: Specifies that the memory implementing the variable must be clocked at twice the rate as the kernel accessing it. This allows for twice as many memory accesses per kernel clock cycle but may reduce the maximum kernel clock frequency.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+ram_stitching
+----
+a|
+If the size of the data is too big for a single RAM to hold it, multiple RAMs  are need to be stitched together. The following values are supported:
+
+* `min_ram`: Specifies that the widths and depths of the individual RAMs may be different to minimize the number of RAMs needed to hold the data. This may result in more complicated logic being implemented to index into the RAMs, reducing the fmax.
+* `max_fmax`: Specifies that the RAMs should be of uniform width and depth, which allows for simple indexing logic to be generated. This may result in more RAMs being used than strictly necessary. 
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+private_copies
+----
+a|
+Specifies that the memory has a defined number of copies to allow simultaneous iterations of a loop at any given time. 
```

Comment:
This seems vague.  Rather than "copies" can we use something like "independent instances" to make it clear that these don't all hold the same data.  This would help to distinguish private_copies from max_replicates (which all hold the same data) below.

It would also be nice if you could make it clear that each of those simultaneous iterations is using a different one of these independent instances.